src/extra/components/pulses.py

```diff
@@ -290,6 +295,7 @@ def get_pulse_mask(self, *args, **kwargs):
              DeprecationWarning, stacklevel=2)
         return self.pulse_mask(*args, **kwargs)
 
+    @lru_cache
```

Comment:
I think we can get away with using [`@cache`](https://docs.python.org/3/library/functools.html#functools.cache) here since the return value will never change.

Reply:
Alrighty, makes sense.